### PR TITLE
#277 Import/export external search settings

### LINKED ISSUE #277: Import/export external search settings when importing/exporting project

### PR DESCRIPTION
**Pull request by az79nefy**
**What's in the PR**
* import and export functionality for a project covers external document repositories
* resolves #277 

**How to test manually**
* create a project with external document repositories
* export project and re-import into new project
* check whether external document repositories in new project match the original project

**Automatic testing**
* [x] PR includes unit tests

**Documentation**
* [ ] PR updates documentation

### CODE REVIEW

File: inception-external-search-core/src/main/java/de/tudarmstadt/ukp/inception/externalsearch/exporter/DocumentRepositoryExporter.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    private final ExternalSearchService externalSearchService;
+    
+    @Autowired
+    public DocumentRepositoryExporter(ExternalSearchService aExternalSearchService)
+    {
+        externalSearchService = aExternalSearchService;
+    }
+    
+    @Override
+    public List<Class<? extends ProjectExporter>> getImportDependencies()
```
Comment (reckart) [inline inception-external-search-core/src/main/java/de/tudarmstadt/ukp/inception/externalsearch/exporter/DocumentRepositoryExporter.java]: I don't think this imported/exporter depends on the LayerExporter.

Comment (az79nefy) [reply inception-external-search-core/src/main/java/de/tudarmstadt/ukp/inception/externalsearch/exporter/DocumentRepositoryExporter.java]: Does this exporter have any dependencies? I could just use the default ``getImportDependencies`` and ``getExportDependencies`` methods provided by the ProjectExporter interface and return an empty list.

**Comment by az79nefy** [comment]
Encountering weird bug: When importing back an exported document, the document repositories in the imported project are correct. But the project that was exported deletes its document repositories, once the same project is imported again.

**Comment by az79nefy** [comment]
Solved bug:

Assigning document repositories an existing ID (even across projects) leads to the deletion of the former document repository.
@reckart I solved this with random number generation for new ID, but this is not fail safe. How do we guarantee unique IDs?

**Comment by reckart** [comment]
When creating/importing a document repository, the ID must be null - the database automatically assigns an ID. If you check the other importers, you'll note that none of them sets the ID field on import and most don't even save the ID field on export.

**Pull request merged**